app/controllers/transactions_controller.rb

```diff
@@ -75,4 +76,8 @@ def transaction_params
   def valid_account?(account)
     account && account.user_id == current_user.id
   end
+
+  def set_query
+    @query = Transactions::TransactionQueryService.new(params).call
```

Comment:
Transaction.. Transaction...
```suggestion
    @query = Transactions::QueryService.new(params).call
```
Acho que entendi a sua abordagem de ter feito um service para querys.
uma pergunta: é realmente necessario fazer buscar por todos esses campos?
para um unico usuário com poucas transações o custo é baixo. 
mas para um banco de dados com milhares de usuarios com milhares de transações isso poderia ser um grande problema. principalmente se existir varios parametros na consulta

da uma olhada nessa gem: https://github.com/activerecord-hackery/ransack

Reply:
Sei que são muitos campos porém eu quis fornecer comodidade em diferentes formas de filtrar os registros das transações.
realmente nao pensei em um banco com muitos usuarios e dados.
mas a quantidade de filtros pode ser reduzida futuramente

Reply:
existe formas mais simples de fazer isso.

outro ponto é `Transactions::TransactionQueryService` até marquei como sugestão, `Transactions::QueryService` uma vez que já esta no namespace (module) de `Transactions` a `QueryService` só pode pertencer a `Transactions`.